manifests/init.pp

```diff
@@ -23,6 +24,12 @@
     fail("Domain name, ${domain}, is invalid.")
   }
 
+  # Validate lookup value
+  validate_array($lookup)
+  if ($lookup != ['UNSET']) and ($::operatingsystem != 'OpenBSD') {
+    fail("\$lookup is only supported on OpenBSD.")
```

Comment:
Could you please use the following fail message. It is inline with my others and provides the full path to the parameter and the detected variable that is not matching which helps in debugging.

``` puppet
fail("the dnsclient::lookup parameter is only supported on OpenBSD. Detected operatingsystem is ${::operatingsystem}.")
```